pkg/registry/core/rest/storage_core.go

```diff
@@ -147,13 +147,13 @@ func New(c Config) (*legacyProvider, error) {
 	return p, nil
 }
 
-func (c *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, error) {
-	apiGroupInfo, err := c.GenericConfig.NewRESTStorage(apiResourceConfigSource, restOptionsGetter)
+func (p *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, error) {
```

Comment:
LGTM, can you crosscheck if all the receivers with `legacyProvider` type are now changed to `p`

Reply:
Yes, I have confirmed that all the receivers with the `legacyProvider` type have been changed to `p`.